OpenRA.Test/OpenRA.Game/MiniYamlTest.cs

```diff
@@ -169,5 +169,30 @@ public void EmptyLinesShouldCountTowardLineNumbers()
 			var result = MiniYaml.FromString(yaml).First(n => n.Key == "TestB");
 			Assert.AreEqual(5, result.Location.Line);
 		}
+
+		[TestCase(TestName = "Duplicated nodes are correctly merged")]
+		public void TestSelfMerging()
+		{
+			var baseYaml = @"
+Test:
+	Merge: original
+		Child: original
+	Original:
+Test:
+	Merge: override
+		Child: override
+	Override:
+";
+
+			var result = MiniYaml.Merge(new[] { baseYaml }.Select(s => MiniYaml.FromString(s, "")));
+			Assert.IsTrue(result.Count(n => n.Key == "Test") == 1, "Result should have exactly one Test node, but does not.");
+
+			var testNodes = result.First(n => n.Key == "Test").Value.Nodes;
+			Assert.IsTrue(testNodes.Count == 3, "Merged Test node should have exactly three children, but has {0}.".F(testNodes.Count));
```

Comment:
Consider `Assert.That(testNodes.Select(n => n.Key), Is.EqualTo(new[] { "Merge", "Original", "Override" }), "Merged Test node has incorrect child nodes.");` to assert the names and ordering are correct as well as the count.

Reply:
> Expected is <System.String[3]>, actual is <System.Linq.Enumerable+SelectListIterator`2[OpenRA.MiniYamlNode,System.String]>

:grin: